lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py

```diff
@@ -202,11 +235,71 @@ def _generate_contig_bin(self, bin_id, file_directory):
 
         return contig_bin
 
+    def _get_contig_file(self, assembly_ref):
+        """
+        _get_contig_file: get contif file from GenomeAssembly object
+        """
+
+        log('retrieving contig file from assembly: {}'.format(assembly_ref))
+        contig_file = self.au.get_assembly_as_fasta({'ref': assembly_ref}).get('path')
+
+        sys.stdout.flush()
+        contig_file = self.dfu.unpack_file({'file_path': contig_file})['file_path']
+
+        log('saved contig file to: {}'.format(contig_file))
+
+        return contig_file
+
+    def _get_contig_string(self, contig_id, assembly_contig_file):
+        """
+        _get_contig_string: find and return contig string from assembly contig file
+        """
+
+        processed_contig_string = False
```

Comment:
I know it's fairly simple to write yourself, but we should still use the BioPython fasta file reader/writer when we can so we don't have to worry about parsing bugs (for example, you can check out what I do here: https://github.com/kbaseapps/AssemblyUtil/blob/master/lib/AssemblyUtil/AssemblyToFasta.py#L87 or here: https://github.com/kbaseapps/AssemblyUtil/blob/master/lib/AssemblyUtil/FastaToAssembly.py#L101).

I forgot to mention that earlier- probably in the uploader you had to read in fasta files as well?  We should use biopython there too.

And since these files can be large, try as much as possible to not load the entire file in memory at once.